tests/test_api_local.py

```diff
@@ -0,0 +1,99 @@
+"""Tests for the local use of multi-omics digital object (modo) API
+"""
+from modo.api import MODO
+from modo.io import build_modo_from_file
+
+import modo_schema.datamodel as model
+import pysam
+import re
+
+## Initialize modo
+
+
+def test_read_modo():
+    MODO("data/ex")
+
+
+def test_init_modo(tmp_path):
+    MODO(tmp_path)
+
+
+def test_init_modo_from_yaml(tmp_path):
+    build_modo_from_file("data/ex_config.yaml", tmp_path)
+
+
+## Add element
+
+
+def test_add_element(Assay, tmp_path):
+    modo = MODO(tmp_path)
+    modo.add_element(Assay)
+    assert "/assay/test_assay" in modo.metadata.keys()
+
+
+def test_add_data(DataEntity, tmp_path):
+    modo = MODO(tmp_path)
+    modo.add_element(DataEntity, data_file="data/ex/demo1.cram")
+    assert "demo1.cram" in [fi.name for fi in modo.list_files()]
+
+
+def test_add_to_parent(Sample, test_modo):
+    test_modo.add_element(Sample, part_of="/assay/assay1")
+    assert "sample/test_sample" in test_modo.metadata["/assay/assay1"].get(
+        "has_sample"
+    )
+
```

Comment:
**note:** Interesting - the sample is actually added with the correct full identifier here.

Reply:
True - it is a bit of a mess. When specifying via `part_of` or `--parent` in the cli the full id is added (it will also only work, if the full id is specified by the user), but when defined via `has_part` property the short id is sufficient. So it differs when specified from the parents or the kids side. I think that is also why subjects appeared to use the full id and objects short id in `modo publish`